manifests/plugin/discovery.pp

```diff
@@ -4,44 +4,51 @@
 #
 # === Parameters:
 #
-# $version::         version string of discovery image, in form of x.y.z-r
+# $major_version::   version string of discovery image, in form of x.y
 #
-# $source::          mirror url from which the image files should be obtained, you
-#                    can use http(s):// or file://
-#
-# $initrd::          name of initrd image file
-#
-# $kernel::          name of kernel file
+# $minor_version::   version string of discovery image, in form of x.y.z
 #
 # $install_images::  should the installer download and setup discovery images
 #                    for you? the average size is few hundreds of MB
 #                    type:boolean
 #
+# $tftp_root::       tftp root to install image into (optional)
+#
+# $source_url::      source URL to download from (optional)
+#
 class foreman::plugin::discovery (
-  $version        = $foreman::plugin::discovery::params::version,
-  $source         = $foreman::plugin::discovery::params::source,
-  $initrd         = $foreman::plugin::discovery::params::initrd,
-  $kernel         = $foreman::plugin::discovery::params::kernel,
+  $major_version  = $foreman::plugin::discovery::params::major_version,
```

Comment:
> > +  $major_version  = $foreman::plugin::discovery::params::major_version,
> 
> This is no longer used.

Right, I am tempted to remove both versions now because in each
directory (started with version 2.0) we have

http://downloads.theforeman.org/discovery/releases/

the fdi-image-latest.tar symlink.

And since we default to "latest" anyway and offer user to change the
base URL, there is no need for this. Much cleaner and more simple.

What you think?

-- 
Later,
 Lukas #lzap Zapletal